addon/components/lmb-plugin/search-modal.ts

```diff
@@ -0,0 +1,157 @@
+import { task as trackedTask } from 'ember-resources/util/ember-concurrency';
+import { LmbPluginConfig } from '@lblod/ember-rdfa-editor-lblod-plugins/plugins/lmb-plugin';
+
+import Mandatee from '@lblod/ember-rdfa-editor-lblod-plugins/models/mandatee';
+import { IBindings } from 'fetch-sparql-endpoint';
+type SearchSort = [keyof Mandatee, 'ASC' | 'DESC'] | false;
+
+interface Args {
+  config: LmbPluginConfig;
+  open: boolean;
+  closeModal: () => void;
+  onInsert: () => void;
+}
+
+interface SparqlResponse {
+  results: {
+    bindings: IBindings[];
+  };
+}
+
+export default class LmbPluginSearchModalComponent extends Component<Args> {
+  // Display
+  @tracked error: unknown;
+  @tracked inputSearchText: string | null = null;
+  @tracked sort: SearchSort = false;
+
+  // Pagination
+  @tracked pageNumber = 0;
+  @tracked pageSize = 20;
+  @tracked totalCount = 0;
+
+  get config() {
+    return this.args.config;
+  }
+
+  @action
+  async closeModal() {
+    await this.servicesResource.cancel();
+    this.args.closeModal();
+  }
+
+  fetchData = restartableTask(async () => {
+    const endpoint = this.args.config.endpoint;
+    const queryResponse = await fetch(endpoint, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        query: `
+        PREFIX besluit: <http://data.vlaanderen.be/ns/besluit#> 
+        PREFIX skos: <http://www.w3.org/2004/02/skos/core#> 
+        PREFIX mandaat: <http://data.vlaanderen.be/ns/mandaat#>
+        PREFIX foaf: <http://xmlns.com/foaf/0.1/>
+        PREFIX persoon: <http://data.vlaanderen.be/ns/persoon#>
+        PREFIX org: <http://www.w3.org/ns/org#>
+        PREFIX regorg: <https://www.w3.org/ns/regorg#>
+        SELECT DISTINCT ?mandatee ?person ?firstName ?lastName ?statusLabel ?fractieLabel ?roleLabel WHERE { 
+            ?mandatee a mandaat:Mandataris;
+              org:holds ?mandaat;
+              mandaat:status ?status;
+              mandaat:isBestuurlijkeAliasVan ?person;
+              org:hasMembership ?membership.
+            ?person foaf:familyName ?lastName;
+              persoon:gebruikteVoornaam ?firstName.
+            ?status skos:prefLabel ?statusLabel.
+            ?membership org:organisation ?fractie.
+            ?fractie regorg:legalName ?fractieLabel.
+            ?mandaat org:role ?role.
+            ?role skos:prefLabel ?roleLabel.
+            OPTIONAL  {
+              ?mandatee mandaat:einde ?endDate
+            }
+            filter (!bound(?endDate) || ?endDate > now()).
+        }        
+        `,
+      }),
+    });
+    const queryJson: SparqlResponse = await queryResponse.json();
+    const mandatees = queryJson.results.bindings.map(Mandatee.fromBinding);
+    return mandatees;
+  });
+
+  search = restartableTask(async () => {
+    // Can't do what I want, so if the user modifies the filter before resolving the query will run again
```

Comment:
what did you want to do?

Reply:
Basically I wanted to avoid refetching the data from the endpoint if the user changes the sort or filter before it finishes, but I couldn't I don't think is really relevant but would be nice to do it.
I don't seem to know enough about ember-concurrency to do what I want

Reply:
As the filter is only applied on the frontend there's no need of refetching the data